### PR TITLE
`Assessment`: Fix an issue with complaints after long assessment periods

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [X] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [X] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Currently, if the assessment due date of an exercise is set, complaints are possible if the result date is not older than the complaint time threshold (default 7 days). If the submission was assessed manually e.g. right after the submission deadline it's not possible to complain if the assessment period is at least the same length as the complaint time threshold or otherwise you have less time

Additionally, it's not quite intuitive to know how many complaints the student has left if it's not possible to complain.

### Description

Now, if the assessment due date is set, it's additionally possible to complain if you are within the time threshold since the end of the assessment period.

I moved the already existing hint up the hierarchy (see screenshots)

### Steps for Testing

**As this works with the complaints threshold which is given in days, it's not possible to test this on the test servers unless you have already existing exercises that suit the requirements or have database access, you want to wait for at least 24h or have access to the database.**
You can use this exercise with **test user 53**: https://artemistest.ase.in.tum.de/courses/515/text-exercises/11107/participate/327958
**Please don't complain but only check if you could when using this exercise.**

Prerequisites:
- 1 Instructor
- 1 Student
- Course with Feedback requests and complaints enabled, the threshold must be **x** days
- 1 Exercise with non-automatic assessment, the assessment due date should be further away than x days from the date the submission as assessed
- 1 Submission of the exercise, handed in properly, assessed at the minimum x days before the assessment due date

Check the exercise as a student, you should still be able to complain.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

complaints-student-view.component.ts: 78.75% | 95.77%
complaints-form.component.ts: 85.71% | 100%

### Screenshots

Before:

After:

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
The code changes look good to me. Thanks for the quick fix!

**Review by MerlinMehmed — DISMISSED** [review]
Tested on TS1, I am able to complain. Code changes look good as well!

**Review by JohannesStoehr — DISMISSED** [review]
Re-approve after the html got moved to another place

**Review by MertAksehirlioglu — DISMISSED** [review]
Tested on TS1, works as described.

File: src/main/webapp/app/complaints/form/complaints-form.component.scss [hunk 1]
```diff
@@ -1,6 +0,0 @@
-.info-icon {
-    color: #ffc107;
-}
```
Comment (FlorianGlombik) [inline src/main/webapp/app/complaints/form/complaints-form.component.scss:3]: Why is this class removed, is it defined at another place? It seems to be used in `complaints-form.component.html`.

**Review by FlorianGlombik — APPROVED** [review]
Tested on ts1 with the prepared submission. The complaint button is shown for testUser53.

I did not quite understand the SCSS change, but I might overlook something there.

**Review by baranimo7 — DISMISSED** [review]
Tested on ts1. Everything works as expected.

File: src/main/webapp/app/complaints/form/complaints-form.component.scss (hunk 1 above)
Comment (julian-christl) [reply src/main/webapp/app/complaints/form/complaints-form.component.scss]: Good catch!

Comment (julian-christl) [reply src/main/webapp/app/complaints/form/complaints-form.component.scss]: Please verify that the icons for complaints and more feedback requests work correct

File: src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts [hunk 2]
```diff
@@ -143,8 +144,12 @@ export class ComplaintsStudentViewComponent implements OnInit {
...
             return false;
         }
-        if (!this.exercise.assessmentDueDate || dayjs().isAfter(dayjs(this.exercise.assessmentDueDate))) {
-            return dayjs().isSameOrBefore(dayjs(completionDate).add(actionThresholdInDays, 'day'));
+        const isWithinThreshold = dayjs().isSameOrBefore(dayjs(completionDate).add(actionThresholdInDays, 'day'));
+        if (!this.exercise.assessmentDueDate) {
+            return isWithinThreshold;
+        }
+        if (dayjs().isAfter(dayjs(this.exercise.assessmentDueDate))) {
+            return isWithinThreshold || dayjs().isBefore(dayjs(this.exercise.assessmentDueDate).add(actionThresholdInDays, 'day'));
```
Comment (JohannesStoehr) [inline src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts:152]: You used isSameOrBefore in 147 but only used isBefore in 152. Is that on purpose?

Comment (julian-christl) [reply src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts]: No, good catch

**Review by JohannesStoehr — APPROVED** [review]
Looked through the code and tested on TS1. Approve 👍

**Review by MerlinMehmed — APPROVED** [review]
Reapproving, change look good and functionality still works as expected

**Review by MertAksehirlioglu — APPROVED** [review]
Tested on TS1, followed the testing steps and works as described.

**Review by FlorianGlombik — APPROVED** [review]
The complain functionality is displayed and the code looks good to me! 👍

**Review by baranimo7 — APPROVED** [review]
tested on ts1. Everything works as expected.

**Pull request merged**